crates/primitives/starknet/src/crypto/hash/pedersen.rs

```diff
@@ -20,10 +21,27 @@ impl HasherT for PedersenHasher {
     /// # Returns
     /// The hash of the data.
     fn hash_bytes(&self, data: &[u8]) -> Felt252Wrapper {
-        // For now we use the first 31 bytes of the data as the field element, to avoid any panics.
-        // TODO: have proper error handling and think about how to hash efficiently big chunks of data.
-        let field_element = FieldElement::from_byte_slice_be(&data[..31]).unwrap();
-        Felt252Wrapper(pedersen_hash(&FieldElement::ZERO, &field_element))
+        // Calculate the number of 31-byte chunks we'll need, rounding up.
+        // (1 byte is used padding to prevent the value of field from being greater than modular)
+        // TODO: It is need a way to truncate bytes to fit into values smaller than modular (optimization)
+        let chunks = (data.len() + 30) / 31;
+
+        let mut hash_value = FieldElement::ZERO;
+
+        for i in 0..chunks {
+            let start = i * 31;
+            let end = cmp::min(start + 31, data.len());
+
+            // Create a buffer for our 32-byte chunk.
+            let mut buffer = [0u8; 32];
+            buffer[1..end - start + 1].copy_from_slice(&data[start..end]);
+
+            // Convert the buffer to a FieldElement and then to a Felt252Wrapper.
+            let field_element = FieldElement::from_bytes_be(&buffer).unwrap();
```

Comment:
How are you sure it is ok to unwrap here?

Reply:
Yes Currently, I have rewritten the code by changing the existing code from_bytes_be to from_byte_slice_be. The `.chunks` function that @EvolveArt pointed out does not exceed the number of field's modulus mentioned in the existing starknet-ff because it divides the chunk into 31bytes. Therefore, it is considered safe.

Reply:
1. Shouldn't we follow some kind of standard for both hashing functions? Is your impl this standard? Or is it something you come up with yourselves?

I'll have to check on that, it's something we implemented because we need to use hashes for danamic bytes. 
However, the hash function shouldn't be a problem if you use the hashes as they are implemented to determine the input value needed for the hash.

2. Isn't it a problem if our Poseidon doesn't behave in the same way that the Starknet one? Or I am missing something?

We have not modified the hash function implemented by Starknet, but we have fixed the existing error when using large input values(bytes).
The existing implementation of Poseidon is as follows, which causes the following error when large bytes are passed.

`let data = FieldElement::from_byte_slice_be(data).unwrap(); `

```
thread 'crypto::hash::poseidon::dynamic_string_hashing' panicked at 'called `Result::unwrap()` on an `Err` value: InvalidLength', 
crates/primitives/starknet/src/crypto/hash/poseidon.rs:21:59
```

So we modified it by separating it into each chunk and hashing them using the existing poseidon_hash_many function.

If you have question, please reply~ Thank you

Reply:
@stompesi, can you comment on your unwrap explaining why it is safe? Or even, when will it stop being safe? So that future developers know this and don't do any mistakes

Reply:
@tdelabro I added a comment on the reason that the unwrap is safe.